pkg/controller/core/clusterqueue_controller.go

```diff
@@ -191,14 +208,41 @@ func (h *cqWorkloadHandler) requestForWorkloadClusterQueue(w *kueue.Workload) *r
 	}
 }
 
+// resourceFlavorHandler enqueues reconcile.Requests in response to ResourceFlavor events.
+// Since the events come from a channel Source, only the Generic handler will
+// receive events.
+type resourceFlavorHandler struct {
```

Comment:
I don't think we need this handler and this whole watch and notification setup. Updating the CQ status and invoking broadcast can be done in the resource flavor controller.